backend/src/layer/trigger/graphql/types/meal.rs

```diff
@@ -0,0 +1,178 @@
+use crate::layer::trigger::graphql::util::{ApiUtil, TRACE_QUERY_MESSAGE};
+use crate::util::MealType;
+use crate::{
+    interface::persistent_data::model,
+    util::{Additive, Allergen, Date, Uuid},
+};
+use async_graphql::{ComplexObject, Context, Result, SimpleObject};
+use tracing::{instrument, trace};
+
+use super::line::Line;
+use super::{image::Image, price::Price, side::Side};
+
+#[derive(SimpleObject, Debug)]
+#[graphql(complex)]
+pub struct Meal {
+    /// The identifier of the main course.
+    id: Uuid,
+    /// The name of the main course.
+    name: String,
+    /// Type of this meal.
+    /// Here the type of meat which is contained in the meal, or whether it is vegetarian or vegan, is specified.
+    meal_type: MealType,
+    /// The ratings given by the users to the meal.
+    ratings: Ratings,
+    /// The prices of the dish each for the four groups of people students, employees, pupils and guests.
+    price: Price,
+    /// Some statistics for the meal.
+    statistics: MealStatistics,
+    #[graphql(skip)]
```

Comment:
the skipped attributes dont need dock comments as they are private and not exposed to the public graphql api.
If they still are wanted to be keept they should be put before the `#` attributes.

Reply:
multiple places